### PR TITLE
[WIP] changed the ansible container bootstrapping

### PR DESCRIPTION
**Pull request by fatz**
- improving the bootstrap process as 25% of DC/OS CI runs where showing docker/docker-ansible bootstrapping issues

### CODE REVIEW

File: bootstrap-docker-install.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    sudo yum install -y --enablerepo $extrasrepo docker
+    sudo systemctl enable docker.service
+    sudo systemctl start docker.service
+    sudo setenforce 0
+    ;;
+  *)
+    echo "installing docker"
+    sudo yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+    sudo yum install -y docker-ce
+    sudo systemctl disable firewalld.service
```
Comment (sebbrandt87) [inline bootstrap-docker-install.sh:29]: maybe a `systemctl daemon-reload` here would help to already be able to call `enable` and `start` for the `docker.service`

File: bootstrap-docker-install.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    OS="$(uname -s)-$(uname -r)"
+fi
+
+echo "checking docker executable"
+if ! [ -x "$(command -v docker)" ]; then
+  case $OS in
+  rhel*)
+    extrasrepo=$(cat /etc/yum.repos.d/redhat* | grep -E '^\[.*extras.*\]$' | grep -vE 'debug|source' | tr -d '[|]')
+    sudo yum install -y --enablerepo $extrasrepo docker
+    sudo systemctl enable docker.service
```
Comment (sebbrandt87) [inline bootstrap-docker-install.sh:19]: maybe a `systemctl daemon-reload` here would help to already be able to call `enable` and `start` for the `docker.service`

File: bootstrap-docker-install.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    sudo systemctl start docker.service
+    sudo setenforce 0
+    ;;
+  *)
+    echo "installing docker"
+    sudo yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
```
Comment (sebbrandt87) [inline bootstrap-docker-install.sh:26]: just to be safe? `which yum-config-manager || sudo yum install -y yum-utils`

**Review by sebbrandt87 — APPROVED** [review]
/lgtm

File: bootstrap-docker-install.sh (hunk 1 above)
Comment (fatz) [reply bootstrap-docker-install.sh:29]: yeah sounds like a good idea

File: bootstrap-docker-install.sh (hunk 3 above)
Comment (fatz) [reply bootstrap-docker-install.sh:26]: kk good point

File: main.tf [hunk 4]
```diff
@@ -78,27 +78,15 @@ resource "null_resource" "run_ansible_from_bootstrap_node_to_install_dcos" {
 
   # with a check for the case that we are on Azure checking the metadata url,
   # as long as cloud-init is not supported on the CentOS and RHEL based azure images
+  provisioner "file" {
+    destination = "/tmp/bootstrap-docker-install.sh"
+
```
Comment (sebbrandt87) [inline main.tf:83]: extra line?

**Review by sebbrandt87 — APPROVED** [review]

File: main.tf (hunk 4 above)
Comment (fatz) [reply main.tf:83]: with only two options we could leave it out sure

File: bootstrap-docker-install.sh [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    sudo yum install -y docker-ce
+    sudo systemctl daemon-reload
+    sudo systemctl disable firewalld.service
+    sudo systemctl stop firewalld.service
+    sudo systemctl enable docker.service
+    sudo systemctl start docker.service
+    sudo setenforce 0
+    ;;
+  esac
+  echo "sleep 30s to let docker become ready"
```
Comment (sebbrandt87) [inline bootstrap-docker-install.sh:36]: we may should do here, to have a timeout just for the case and not sleep that long if dockerd is not ready yet.
```
timeout -t 30 bash <<EOF
while true; do
  docker info >/dev/null 2>&1
  if [ $? -eq 0 ]; then
    break
  fi
  sleep 10
done
EOF
```

**Pull request merged**